### PR TITLE
Enable QSPI tests on ESP32 and clean up

### PR DESCRIPTION
**Pull request by bugadani**
I don't think I've done any behavioural changes to esp-hal to make this work.

### CODE REVIEW

File: esp-hal/src/spi/master.rs [hunk 1]
```diff
@@ -2483,34 +2413,22 @@ pub trait Instance: private::Sealed {
...
-                .cs_hold()
-                .set_bit()
-                .usr_dummy_idle()
-                .set_bit()
-                .usr_addr()
-                .clear_bit()
-                .usr_command()
-                .clear_bit()
+            w.usr_miso_highpart().clear_bit();
+            w.usr_miso_highpart().clear_bit();
```
Comment (Dominaezzz) [inline esp-hal/src/spi/master.rs:2417]:
```suggestion
            w.usr_mosi_highpart().clear_bit();
```

Comment (bugadani) [reply esp-hal/src/spi/master.rs:2417]: Nice, this indicates an advantage of this style :D

File: esp-hal/src/spi/master.rs [hunk 2]
```diff
@@ -3185,36 +2931,23 @@ pub trait Instance: private::Sealed {
...
-                .usr_dummy_idle()
-                .bit(dummy_idle)
-                .usr_dummy()
-                .bit(dummy_state)
-                .usr_addr()
-                .bit(address_state)
-                .usr_command()
-                .bit(command_state)
+            w.usr_miso_highpart().clear_bit();
+            w.usr_miso_highpart().clear_bit();
```
Comment (bugadani) [inline esp-hal/src/spi/master.rs:2935]:
```suggestion
            w.usr_mosi_highpart().clear_bit();
```

**Review by bjoernQ — APPROVED** [review]
LGTM

**Review by MabezDev — APPROVED** [review]

**Pull request merged**